### ISSUE TITLE
Fail to import RTCMultiConnection (npm) on Ionic 3 App

### ISSUE THREAD

**Issue by andreportaro** [muaz-khan/RTCMultiConnection#481]
Hey there,
I'm having trouble importing the package (https://www.npmjs.com/package/rtcmulticonnection-v3) into a Ionic app.
How could I include it in the "import" style, so I can have a .ts module make the new RTCMultiConnection() statement?

Thanks!!

**Comment by drsss** [comment]
hello,

you need to declare RTCMultiConnection in the declaration file.

`declare module 'RTCMultiConnection';`

then you can use in RTCMulticonnection in any ts file.

**Comment by felix9607** [comment]
`<script src="https://rtcmulticonnection.herokuapp.com/dist/RTCMultiConnection.min.js"></script>
  <script src="https://rtcmulticonnection.herokuapp.com/socket.io/socket.io.js"></script>`

`declare var RTCMultiConnection: any;

@IonicPage()
@Component({`

I did sth like this and it seems to work actually, but I got the Problem, that it doesn't Show a Picture. So now I don't know where it's coming from.
Do you think it's using the js correctly?

**Comment by felix9607** [comment]
@drsss can you give me a better example please? Do you mean the 'app.module.ts' file with "declaration file"?

right now I'm doing this:
`<script src="https://rtcmulticonnection.herokuapp.com/dist/RTCMultiConnection.min.js"></script>
  <script src="https://rtcmulticonnection.herokuapp.com/socket.io/socket.io.js"></script>`

and

`declare var RTCMultiConnection: any;` but I'm not sure if this way works 100%.

**Comment by drsss** [comment]
@felix9607 yes, you can also declare module for RTCMultiConnection in app.module.ts file

**Comment by felix9607** [comment]
I'm kinda lost.

Can you please tell me what to exactly do, to Import RTCMulticonnection correctly?

I installed the module via npm `npm i rtcmulticonnection-v3`

If I try `declare module 'RTCMultiConnection';` in app.module.ts it says: 
"Couldn't find module 'RTCModuleConnection'"
but there is the Folder `rtcmulticonnection-v3` in @node_modules

**Comment by drsss** [comment]
@felix9607 for work around can you try to remove reference of RTCMultiConnection then run ionic serve

after successfully run add reference for RTCMultiConnection

and use 
<script src="https://rtcmulticonnection.herokuapp.com/dist/RTCMultiConnection.min.js"></script>
<script src="https://rtcmulticonnection.herokuapp.com/socket.io/socket.io.js"></script>

**Comment by felix9607** [comment]
That's what I'm doing allready. But I wanted to use the modules, to get Code suggestions and so on.
Another reason is, that in his examples he uses functions like this:
`connection.selectDevices(audioInputDevices.value);`
`connection.selectDevices(videoOutputDevices.value);`

But selectDevices doesn't exist in this js files?
so I thought i should try the modules.

**Comment by muaz-khan** [comment]
v3 allows you select cameras/mics as following:

```javascript
connection.mediaConstraints = {
    audio: {
        deviceId: DetectRTC.audioInputDevices[0].deviceId
    },
    video: {
        deviceId: DetectRTC.videoInputDevices[0].deviceId
    }
};
```

Now you can use `openOrJoin` or `captureUserMedia+renegotiate` or `addStream`.

```javascript
// during a live session
connection.addStream({
     video: true,  // change only camera, you can ignore mic
     oneway: true // make sure that addStream action applies only on your peer
});
```

**Comment by rafaj10** [comment]
Guys i need help with this as well,  I'm trying to make it work with ionic 3 but I can't ..., can someone share some base code?

**Comment by rafaj10** [comment]
This is my code:

import { Component } from '@angular/core';
import { NavController } from 'ionic-angular';

export declare var RTCMultiConnection: any;

@Component({
  selector: 'page-home',
  templateUrl: 'home.html',
})

export class HomePage {

  connection = new RTCMultiConnection();

  constructor(public navCtrl: NavController) {

    this.connection.socketURL = 'https://its4-conference.herokuapp.com/';

    this.connection.openOrJoin('roomid');

    this.connection.onstream = function (e) {
      document.getElementById("video_div").appendChild(event.mediaElement);
      //console.log(event.mediaElement);
    };
  }

}


but the result on iOS is, 

2017-11-23 16:31:08.144010-0200 MyApp[1085:392403] ERROR: ERROR TypeError: undefined is not a function (near '...navigator.getUserMedia...')

**Comment by hardcorelite** [comment]
@rafaj10 in the end you managed to do it?

**Comment by lucky72o** [comment]
@rafaj10 or @hardcorelite or @felix9607 
Is anyone of you managed to do this? Is there any simple example with ionic 2 of how to make it work?

**Comment by hardcorelite** [comment]
@lucky72o yes what exactly at you it is impossible?

**Comment by lucky72o** [comment]
@hardcorelite 

I’m quite new to ionic and I’m struggling to understand how to integrate RTCMultiConnection into ionic in general.

I’m trying to follow http://www.rtcmulticonnection.org/docs/Write-Android-Apps/ example

I’ve added scripts to index.html

Now in one of my .ts files I’m trying to create a new connection:
var connection = new RTCMultiConnection();

But I don’t understand how can I import RTCMultiConnection in ionic way …

**Comment by hardcorelite** [comment]
`declare var RTCMultiConnection: any;`
`export class ContactPage {
  connection = new RTCMultiConnection();.......`
and your index.html you need to register the scripts to the files
